### PR TITLE
11385 naics service (api call) + validation

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#11385

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/naics.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+class NaicsService:
+    """Provides services to use the naics-api."""
+
+    @staticmethod
+    def find_by_code(naics_code: str):
+        """Return NAICS Structure matching code."""
+        naics_url = current_app.config.get('NAICS_API_URL')
+        token = AccountService.get_bearer_token()
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/naics.py:31]: Using `system` token so that this can be called from both api and filer

**Review by lekshmimallika-aot — APPROVED** [review]

**Comment by argush3** [comment]
the filer changes will be in a different PR right?

File: legal-api/src/legal_api/services/filings/validations/registration.py [hunk 2]
```diff
@@ -83,6 +84,16 @@ def validate_business_type(filing: Dict, legal_type: str) -> list:
     return msg
 
 
+def validate_naics(filing: Dict, filing_type='registration') -> list:
+    """Validate naics."""
+    msg = []
+    naics_code_path = f'/filing/{filing_type}/business/naics/naicsCode'
+    if (naics_code := get_str(filing, naics_code_path)) and not NaicsService.find_by_code(naics_code):
```
Comment (argush3) [inline legal-api/src/legal_api/services/filings/validations/registration.py:91]: I think we should validation that the description in the filing matches up to the value pulled back from the NAICS endpoint too.

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/registration.py:91]: ok will do that

**Review by argush3 — APPROVED** [review]

**Pull request merged**